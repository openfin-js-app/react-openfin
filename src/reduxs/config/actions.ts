import {createAction, ActionFunctionAny, Action} from 'redux-actions';

import {
    IConfigDexie,
    IConfigResetOption,
    IConfigUpdateOneFieldOption,
    IConfigUpdateGlobalFilterStrOption,
    IConfigDoUpdateOneFieldOption
} from './types';

export const CONFIG_RESET:string = 'CONFIG_RESET';

// redux
export const CONFIG_DO_UPDATE_ONE_FIELD:string = 'CONFIG_DO_UPDATE_ONE_FIELD';
export const CONFIG_UPDATE_GLOBAL_FILTER_STR:string = 'CONFIG_UPDATE_GLOBAL_FILTER_STR';
export const configReset:ActionFunctionAny<Action<IConfigResetOption>>
    = createAction(CONFIG_RESET, (option)=>(option));
export const configDoUpdateOneField:ActionFunctionAny<Action<IConfigDoUpdateOneFieldOption>>
    = createAction(CONFIG_DO_UPDATE_ONE_FIELD, (option)=>(option));
export const configUpdateGlobalFilterStr:ActionFunctionAny<Action<IConfigUpdateGlobalFilterStrOption>>
    = createAction(CONFIG_UPDATE_GLOBAL_FILTER_STR, (option)=>(option));

// saga
export const CONFIG_LOAD_FROM_DEXIE:string = 'CONFIG_LOAD_FROM_DEXIE';
export const configLoadFromDexie:ActionFunctionAny<Action<void>>
    = createAction(CONFIG_LOAD_FROM_DEXIE,);
export const CONFIG_DO_UPDATE_ONE_FIELD_IN_DEXIE:string = 'CONFIG_DO_UPDATE_ONE_FIELD_IN_DEXIE';
export const configDoUpdateOneFieldInDexie:ActionFunctionAny<Action<IConfigDexie>>
    = createAction(CONFIG_DO_UPDATE_ONE_FIELD_IN_DEXIE, (option)=>(option));

export const CONFIG_UPDATE_ONE_FIELD:string = 'CONFIG_UPDATE_ONE_FIELD';
export const configUpdateOneField:ActionFunctionAny<Action<IConfigUpdateOneFieldOption>>
    = createAction(CONFIG_UPDATE_ONE_FIELD, (option)=>(option));



// optional actions depending on the config application values

export const CONFIG_UPDATE_NEW_WINDOW_POSITION:string = 'CONFIG_UPDATE_NEW_WINDOW_POSITION';
export const CONFIG_UPDATE_NEW_WINDOW_POSITION_ADD_DELTA:string = 'CONFIG_UPDATE_NEW_WINDOW_POSITION_ADD_DELTA';
export const CONFIG_UPDATE_NEW_WINDOW_POSITION_RESET_TOP:string = 'CONFIG_UPDATE_NEW_WINDOW_POSITION_RESET_TOP';
export const CONFIG_UPDATE_NEW_WINDOW_POSITION_RESET_LEFT:string = 'CONFIG_UPDATE_NEW_WINDOW_POSITION_RESET_LEFT';
export const configUpdateNewWindowPosition:ActionFunctionAny<Action<void>>
    = createAction(CONFIG_UPDATE_NEW_WINDOW_POSITION);
export const configUpdateNewWindowPositionAddDelta:ActionFunctionAny<Action<void>>
    = createAction(CONFIG_UPDATE_NEW_WINDOW_POSITION_ADD_DELTA);
export const configUpdateNewWindowPositionResetTop:ActionFunctionAny<Action<void>>
    = createAction(CONFIG_UPDATE_NEW_WINDOW_POSITION_RESET_TOP);
export const configUpdateNewWindowPositionResetLeft:ActionFunctionAny<Action<void>>
    = createAction(CONFIG_UPDATE_NEW_WINDOW_POSITION_RESET_LEFT);