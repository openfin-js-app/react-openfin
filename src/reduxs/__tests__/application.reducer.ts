import {Docking, Event, System, Window} from 'redux-openfin';
import * as Actions from '../application/actions';
import reducerCreator, {defaultState} from '../application/reducer';


describe('Application reducer',()=>{

    describe('With parent state', () => {

        let reducer

        beforeAll(()=>{
            reducer = reducerCreator(defaultState);
        })


        it('System.actions.GET_MACHINE_ID_RES reduced correctly',()=>{
            const action:any = {type:System.actions.GET_MACHINE_ID_RES, payload:{id:'id'}};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });


        it('System.actions.GET_DEVICE_USER_ID_RES reduced correctly',()=>{
            const action:any = {type:System.actions.GET_DEVICE_USER_ID_RES, payload:{id:'id'}};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });


        it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - username',()=>{
            const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                    env:'username', value:'usernameVal'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - computername',()=>{
            const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                    env:'computername', value:'computernameVal'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

    });

    describe('With out parent state', () => {

        let reducer

        beforeAll(()=>{
            reducer = reducerCreator();
        })

        it('System.actions.GET_MACHINE_ID_RES reduced correctly',()=>{
            const action:any = {type:System.actions.GET_MACHINE_ID_RES, payload:{id:'id'}};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });


        it('System.actions.GET_DEVICE_USER_ID_RES reduced correctly',()=>{
            const action:any = {type:System.actions.GET_DEVICE_USER_ID_RES, payload:{id:'id'}};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });


        it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - username',()=>{
            const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                    env:'username', value:'usernameVal'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - computername',()=>{
            const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                    env:'computername', value:'computernameVal'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - unmatched',()=>{
            const action:any = {type:System.actions.GET_ENVIRONMENT_VARIABLE_RES, payload:{
                    env:'someEnv', value:'someVal'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('System.actions.GET_VERSION_RES reduced correctly',()=>{
            const action:any = {type:System.actions.GET_VERSION_RES, payload:{
                    version:'versionStr'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('System.actions.GET_HOST_SPECS_RES reduced correctly',()=>{
            const action:any = {type:System.actions.GET_HOST_SPECS_RES, payload:{
                    type: 'GET_HOST_SPECS_RES'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('Window.actions.GET_STATE_RES reduced correctly',()=>{
            const action:any = {type:Window.actions.GET_STATE_RES, payload:{
                    state:'windowState'
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('Window.actions.GET_BOUNDS_RES reduced correctly',()=>{
            const action:any = {type:Window.actions.GET_BOUNDS_RES, payload:{
                    top:0,
                    left:1,
                    right:2,
                    height:3,
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('Event.actionDicts.windowEventDictByName[\'bounds-changing\'].type reduced correctly',()=>{
            const action:any = {type:Event.actionDicts.windowEventDictByName['bounds-changing'].type, payload:{
                    top:0,
                    left:1,
                    right:2,
                    height:3,
                }};
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        describe('Event.actionDicts.windowEventDictByName[\'group-changed\'].type action',  () => {

            const windowsName = 'windowsName';

            beforeAll(()=>{
                window.name = windowsName;
            })

            it('join reason and correct window name',()=>{
                const action:any = {type:Event.actionDicts.windowEventDictByName['group-changed'].type, payload:{
                        sourceWindowName:windowsName,
                        reason:Docking.types.GroupEventReason.JOIN,
                        targetWindowName:"",
                        memeberOf:"",
                    }};
                expect(reducer(undefined,action)).toMatchSnapshot();
            })

            it('leave reason and correct window name',()=>{
                const action:any = {type:Event.actionDicts.windowEventDictByName['group-changed'].type, payload:{
                        sourceWindowName:windowsName,
                        reason:Docking.types.GroupEventReason.LEAVE,
                        targetWindowName:"",
                        memeberOf:"",
                    }};
                expect(reducer(undefined,action)).toMatchSnapshot();
            })

            it('disband reason and correct window name',()=>{
                const action:any = {type:Event.actionDicts.windowEventDictByName['group-changed'].type, payload:{
                        sourceWindowName:"",
                        reason:Docking.types.GroupEventReason.DISBAND,
                        targetWindowName:"",
                        memeberOf:"",
                    }};
                expect(reducer(undefined,action)).toMatchSnapshot();
            })

        });

        it('APPLICATION_UPDATE_DOCK_STATUS reduced correctly',()=>{
            const action:any = Actions.applicationUpdateDockStatus(true);
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_SET_LOADING_MSG reduced correctly',()=>{
            const action:any = Actions.applicationSetLoadingMsg('str');
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_READY reduced correctly',()=>{
            const action:any = Actions.applicationReady({});
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_CHILD_READY reduced correctly',()=>{
            const action:any = Actions.applicationChildReady({});
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_NOTIFICATION_READY reduced correctly',()=>{
            const action:any = Actions.applicationNotificationReady({});
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_DRAWER_TOGGLE reduced correctly',()=>{
            const action:any = Actions.applicationDrawerToggle();
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_NEW_SNACKBAR reduced correctly',()=>{
            const action:any = Actions.applicationNewSnackbar({
                message:'Snackbar message',
                variant:'primary',
            });
            const resultState = reducer(undefined,action);
            delete resultState.snackBarMsgQueue[0].key;
            expect(resultState).toMatchSnapshot();
        });

        it('APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - nonempty',()=>{
            const action:any = Actions.applicationProcessSnackbarQueue();
            expect(reducer({
                snackBarOpen:false,
                snackBarMsgInfo:{},
                snackBarMsgQueue:[{
                    message:'Snackbar message',
                    variant:'primary',
                }]
            },action)).toMatchSnapshot();
        });

        it('APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - empty',()=>{
            const action:any = Actions.applicationProcessSnackbarQueue();
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_SET_SNACKBAR_STATUS reduced correctly - empty',()=>{
            const action:any = Actions.applicationSetSnackbarStatus(true);
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE reduced correctly - empty',()=>{
            const action:any = Actions.applicationLaunchBarToggleCollapse();
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_NETWORK_ONLINE reduced correctly - empty',()=>{
            const action:any = Actions.applicationNetworkOnline();
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

        it('APPLICATION_NETWORK_OFFLINE reduced correctly - empty',()=>{
            const action:any = Actions.applicationNetworkOffline();
            expect(reducer(undefined,action)).toMatchSnapshot();
        });

    });

});
