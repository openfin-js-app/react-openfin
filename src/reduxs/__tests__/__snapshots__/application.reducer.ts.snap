// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application reducer With out parent state APPLICATION_CHILD_READY reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": false,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_DRAWER_TOGGLE reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": false,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_LAUNCH_BAR_TOGGLE_COLLAPSE reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": true,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_NETWORK_OFFLINE reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": true,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_NETWORK_ONLINE reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_NEW_SNACKBAR reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [
    Object {
      "message": "Snackbar message",
      "variant": "primary",
    },
  ],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_NOTIFICATION_READY reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": false,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_PROCESS_SNACKBAR_QUEUE reduced correctly - nonempty 1`] = `
Object {
  "snackBarMsgInfo": Object {
    "message": "Snackbar message",
    "variant": "primary",
  },
  "snackBarMsgQueue": Array [],
  "snackBarOpen": true,
}
`;

exports[`Application reducer With out parent state APPLICATION_READY reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": false,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_SET_LOADING_MSG reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "str",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_SET_SNACKBAR_STATUS reduced correctly - empty 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": true,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state APPLICATION_UPDATE_DOCK_STATUS reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": true,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state Event.actionDicts.windowEventDictByName['bounds-changing'].type reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 3,
  "winLeft": 1,
  "winTop": 0,
  "winWidth": undefined,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state Event.actionDicts.windowEventDictByName['group-changed'].type action disband reason and correct window name 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state Event.actionDicts.windowEventDictByName['group-changed'].type action join reason and correct window name 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": true,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state Event.actionDicts.windowEventDictByName['group-changed'].type action leave reason and correct window name 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_DEVICE_USER_ID_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": "id",
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - computername 1`] = `
Object {
  "computerName": "computernameVal",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - unmatched 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - username 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "usernameVal",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_HOST_SPECS_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {
    "type": "GET_HOST_SPECS_RES",
  },
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_MACHINE_ID_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": "id",
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state System.actions.GET_VERSION_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "versionStr",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state Window.actions.GET_BOUNDS_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 3,
  "winLeft": 1,
  "winTop": 0,
  "winWidth": undefined,
  "windowsState": "normal",
}
`;

exports[`Application reducer With out parent state Window.actions.GET_STATE_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "windowState",
}
`;

exports[`Application reducer With parent state System.actions.GET_DEVICE_USER_ID_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": "id",
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With parent state System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - computername 1`] = `
Object {
  "computerName": "computernameVal",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With parent state System.actions.GET_ENVIRONMENT_VARIABLE_RES reduced correctly - username 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": null,
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "usernameVal",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;

exports[`Application reducer With parent state System.actions.GET_MACHINE_ID_RES reduced correctly 1`] = `
Object {
  "computerName": "",
  "deviceUserId": null,
  "docked": false,
  "drawerOpen": true,
  "launchBarCollapse": false,
  "loading": true,
  "loadingMsg": "",
  "machineId": "id",
  "offline": false,
  "openfinHostSpec": Object {},
  "openfinVersion": "n/a",
  "snackBarMsgInfo": Object {},
  "snackBarMsgQueue": Array [],
  "snackBarOpen": false,
  "username": "",
  "winHeight": 0,
  "winLeft": 0,
  "winTop": 0,
  "winWidth": 0,
  "windowsState": "normal",
}
`;
